### ISSUE TITLE
Support libraries in non node_modules folders

### ISSUE THREAD

**Issue by ThomasKerbrat** [angular/angular#16281]
**I'm submitting a ...**  (check one with "x")
```
[x] bug report
```

**Current behavior**
I have a setup with two local NPM packages, they are not published on npmjs and are not intended to be. I use the packages to benefit from the [local paths](https://docs.npmjs.com/files/package.json#local-paths) dependencies.
One package is a library which declares reusable components, modules, and services and the other package is an application (created with `ng new`) that uses the library.
When the application imports the library (es2015 import, then the `imports` array in `@NgModule` decorator) I have the following error at compile time:
`ERROR in DemoLibModule is not an NgModule`

**Expected behavior**
I expect that the `DemoLibModule` module to be recognized as an NgModule.

**Minimal reproduction of the problem with instructions**
I made a GitHub repository. Please see: https://github.com/ThomasKerbrat/didactic-enigma
Where I import my library into the app: https://github.com/ThomasKerbrat/didactic-enigma/blob/master/my-demo-app/src/app/app.module.ts#L18
Where I declare my DemoLibModule: https://github.com/ThomasKerbrat/didactic-enigma/blob/master/my-demo-lib/src/demo-lib.module.ts#L13

**What is the motivation / use case for changing the behavior?**
I'm maybe doing something wrong, but I couldn't find documentation for my use case.

**Please tell us about your environment:**
Debian 9, VS Code, NPM

* **Angular version:** 4.0.3

* **Browser:** N/A
 
* **Language:** TypeScript 2.2.2

* **Node (for AoT issues):**
`node --version` = v7.9.0
`npm --version` = 4.2.0

Best regards,

**Comment by tbosch** [comment]
Thanks for reporting. `ngc` does not support libraries in non `node_modules` folders right now. See also https://github.com/angular/angular/issues/12249.

**Comment by ThomasKerbrat** [comment]
Hello, thank you for your response.
I have updated the reproduction repository with new information I gathered since I posted this issue.

The original issue `ERROR in DemoLibModule is not an NgModule` was resolved by this commit:
[7b0cae583020def04d958f06cbe7004a95c940c8](https://github.com/ThomasKerbrat/didactic-enigma/commit/7b0cae583020def04d958f06cbe7004a95c940c8), specifically the files [my-demo-lib/package.json](https://github.com/ThomasKerbrat/didactic-enigma/commit/7b0cae583020def04d958f06cbe7004a95c940c8#diff-61ede9a4d808743890d68f30cc68a082) and [my-demo-lib/tsconfig.json](https://github.com/ThomasKerbrat/didactic-enigma/commit/7b0cae583020def04d958f06cbe7004a95c940c8#diff-feeac0e4d2ea95ace6ac52c0ec5de4ef).

Basically, I was not compiling the library with `ngc`.

I have now an other error message, maybe unrelated to the original one, and seems to be related to #15767.
Thank you for your time, have a nice day :)
